feat: Add pixi to the template

YurelyCamacho opened the pull request:
## Pull Request description

Edit and create files needed to add pixi to the template

Fixes #262 

## Pull Request checklists

This PR is a:

- [ ] bug-fix
- [X] new feature
- [ ] maintenance

About this PR:

- [ ] it includes tests.
- [ ] the tests are executed on CI.
- [ ] the tests generate log file(s) (path).
- [X] pre-commit hooks were executed locally.
- [X] this PR requires a project documentation update.

xmnlab commented on src/scicookie/{{cookiecutter.project_slug}}/.makim.yaml:
```diff
@@ -115,6 +115,8 @@ groups:
           {%- elif cookiecutter.build_system == "pybind11" %}
           python -m build
+          {%- elif cookiecutter.build_system == "pixi" %}
+          cargo build
```
My impression is that it should be pixi install

xmnlab commented on src/scicookie/{{cookiecutter.project_slug}}/Makefile:
```diff
@@ -129,6 +129,8 @@ build:
 	{%- elif cookiecutter.build_system == "pybind11" %}
 	python -m build
+	{%- elif cookiecutter.build_system == "pixi" %}
+	cargo build
```
Also here, i guess that it should be pixi install ..

xmnlab commented:
https://github.com/osl-incubator/scicookie/actions/runs/10016410500/job/27689312914#step:5:5566

Not sure why it is not finding pyproject.toml .. maybe it is missing som pixi table in the pyproject.toml

YurelyCamacho commented:
@xmnlab Can you help me with this please.

xmnlab commented:
@YurelyCamacho , this is the error in the log:

`/home/runner/work/scicookie/scicookie/tests/smoke/base.sh: line 102: ipython: command not found`

that happens because pixi doesn't install the dependencies in the current environment, it creates an isolated environment for that.

so you need to activate the environment first or execute it with the pixi run command: 

* https://pixi.sh/latest/features/environment/#activation
* https://pixi.sh/latest/features/environment/#traditional-conda-activate-like-activation


> It is not encouraged to use the traditional conda activate-like activation, as deactivating the environment is not really possible. 

so I would go with `pixi run` approach. In the past we used something similar for `hatch`, so you can use the same approach: https://github.com/osl-incubator/scicookie/commit/32704a5fba35869f5a67c6b7db76f933e9abd211#diff-1c8f1585e21b07c45bdfaa5e9e131567e4c361fbae1a7fc2498a60ba061d70ddR43

YurelyCamacho commented:
@xmnlab I used the approach you told me about in the previous comment and the same error remains. I also changed your suggestion of `cargo build` to `pixi install` and also `python -m build` and it didn't work. I made other changes based on hatch and it doesn't work either (for example, I changed `pixi install` to pip install “.[dev]” in the base.sh).

YurelyCamacho commented:
@xmnlab I can't run the tests locally because I keep getting the error I told you before about mamba, and I have it installed in the environment

xmnlab commented:
> @xmnlab I can't run the tests locally because I keep getting the error I told you before about mamba, and I have it installed in the environment

let me know if you want a quick call to check that together.

about the current error on ci, the COMMAND_PREFIX is just to run the other commands with pixi run ... but you still need to install the dependencies .. so maybe with pip install should do the trick .. so you need to have both

YurelyCamacho commented:
@xmnlab Yes, tomorrow would be good to verify that. What time can you do it?
I'll see if what you told me works, thanks.

YurelyCamacho commented:
@xmnlab I think the problem is in the [tool.pixi.dependencies]. I will leave the prefix.dev answer here.

Hey! The `[pypi-dependencies]` part doesn't do anything as for pixi to understand it you would need to add `tool.pixi` to it. 

I see you added `build` to the `dev` optional dependencies. If you want to install a dev env with pixi you need to add:
```
[environments]
dev = ["dev"]
```
then you should be able to do `pixi run -e dev python -m build`

xmnlab commented on tests/smoke/base.sh:
```diff
@@ -54,23 +58,23 @@ else
 fi
 
-python -c "import osl_python_package as mypkg; assert mypkg.__version__ == '0.1.0'"
+$COMMAND_PREFIX python -c "import osl_python_package as mypkg; assert mypkg.__version__ == '0.1.0'"
```
not sure .. but it looks like that there is any issue running this with pixi ... so maybe you could try a different approach:

```suggestion
$COMMAND_PREFIX python -c 'import osl_python_package as mypkg; assert mypkg.__version__ == "0.1.0"'
```

xmnlab commented:
LGTM! Good job, @YurelyCamacho ! Thanks for working on that
